packages/config/test/index.spec.ts

```diff
@@ -4,6 +4,7 @@ import sinon from 'sinon'
 import sinonChai from 'sinon-chai'
 
 import * as configUtil from '../src/index'
+import { OverrideLevel, ALL_OVERRIDE_LEVELS } from '../src/options'
```

Comment:
Here is a neat TS trick you could add to `src/options.ts` to reduce some duplication: https://github.com/cypress-io/cypress/compare/test-isolation-config-opt-2...lmiller/test-isolation-config-opt-2-ts-update?expand=1

The trick is

```ts
const abc = ['a', 'b', 'c'] as const

type ABC = typeof abc[number] // 'a' | 'b' | 'c'
```

By casting the array `as const`, the array becomes immutable, so TS can infer that only `'a' | 'b' | 'c'` is valid. If you remove `const`, it'll become `Array<string>` instead of `Array<('a' | 'b' | 'c')>`.

Reply:
I was trying to do something like this, but I'm def still a newbie at typescript. With your suggestion, I'm seeing the following: 

<img width="1304" alt="Screen Shot 2022-08-02 at 7 28 23 PM" src="https://user-images.githubusercontent.com/14099737/182499178-8729c0b4-d10c-4b14-9aca-47d0387dcd31.png">
<img width="1297" alt="Screen Shot 2022-08-02 at 7 28 04 PM" src="https://user-images.githubusercontent.com/14099737/182499179-0982b914-4760-415c-8621-4b3d67de12a4.png">


I don't think did it quite right 😅

Reply:
ah wow. re-reading, you literally provided the code....thought `typeof abc[number]` was a typo where number was meant to be string. I need to pay closer attentin.

Reply:
Odd, seems fine for me? https://app.circleci.com/pipelines/github/cypress-io/cypress/41239/workflows/62972cbb-f8c1-4486-baaf-27ea5c645ef5/jobs/1706374?invite=true#step-114-7

I think you missed `Readonly`

```ts
export type OverrideLevels = Readonly<Array<typeof ALL_OVERRIDE_LEVELS[number]> | 'never'>
                                               ^^^^^^^^^ this
```

Reply:
https://github.com/cypress-io/cypress/pull/23040/commits/191105ba94ac63d4e1ed69e02bc83d3beab9ec12